crates/bindings/src/lib.rs

```diff
@@ -45,13 +42,610 @@ pub use timestamp::Timestamp;
+/// ```text
+/// ctx.db.{table}().{column}()
+/// ```
+/// to get a [`BTreeIndex`] from a [`ReducerContext`].
+///
+/// For example:
+///
+/// ```ignore
+/// ctx.db.cities().latitude()
+/// ```
+///    
+/// # Generated code
+///
+/// For each `[table(name = {name})]` annotation on a type `{T}`, generates a struct
+/// `{name}__TableHandle` implementing [`Table<Row={T}>`](crate::Table), and a trait that allows looking up such a
+/// `{name}Handle` in a [`ReducerContext`].
+///
+/// The struct `{name}__TableHandle` is public and lives next to the row struct.
+///
+/// For each named index declaration, add a method to `{name}__TableHandle` for getting a corresponding
+/// [`BTreeIndex`].
+///
+/// For each field  with a `#[unique]` or `#[primary_key]` annotation,
+/// add a method to `{name}Handle` for getting a corresponding [`UniqueColumn`].
+///
+/// The following pseudocode illustrates the general idea. Curly braces are used to indicate templated
+/// names.
+///
+/// ```ignore
+/// use spacetimedb::{BTreeIndex, UniqueColumn, Table, DbView};
+///
+/// // This generated struct is hidden and cannot be directly accessed.
+/// struct {name}__TableHandle { /* ... */ };
+///
+/// // It is a table handle.
+/// impl Table for {name}__TableHandle {
+///     type Row = {T};
+///     /* ... */
+/// }
+///
+/// // It can be looked up in a `ReducerContext`,
+/// // using `ctx.db().{name}()`.
+/// trait {name} {
+///     fn {name}(&self) -> Row = {T}>;
+/// }
+/// impl {name} for <ReducerContext as DbContext>::DbView { /* ... */ }
+///
+/// // Once looked up, it can be used to look up indexes.
+/// impl {name}Handle {
+///     // For each `#[unique]` or `#[primary_key]` field `{field}` of type `{F}`:
+///     fn {field}(&self) -> UniqueColumn<_, {F}, _> { /* ... */ };
+///     
+///     // For each named index `{index}` on fields of type `{(F1, ..., FN)}`:
+///     fn {index}(&self) -> BTreeIndex<_, {(F1, ..., FN)}, _>;
+/// }
+/// ```
+///
+/// [`Table<Row = Self>`]: `Table`
+#[doc(inline)]
+pub use spacetimedb_bindings_macro::table;
+
+/// Marks a function as a spacetimedb reducer.
+///
+/// A reducer is a function with read/write access to the database
+/// that can be invoked remotely by [clients].
+///
+/// Each reducer call runs in its own database transaction,
+/// and its updates to the database are only committed if the reducer returns successfully.
+///
+/// The first argument of a reducer is always a [`&ReducerContext`]. This context object
+/// allows accessing the database and viewing information about the caller, among other things.
+///
+/// After this, a reducer can take any number of arguments.
+/// These arguments must implement the [`SpacetimeType`], [`Serialize`], and [`Deserialize`] traits.
+/// All of these traits can be derived at once by marking a type with `#[derive(SpacetimeType)]`.
+///
+/// Reducers may return either `()` or `Result<(), E>` where `E: Debug`.
+///
+/// ```rust,ignore
```

Comment:
Why is this `ignore`? In general, why are so many of the code blocks in your docs marked `ignore`? I fear that ignored code blocks are doomed to bitrot and become incorrect, which is why Rustdoc supports doctests.

Reply:
Fixed